packages/shared/src/variant.ts

```diff
@@ -1,22 +1,51 @@
 import { AbstractGame } from "./abstract_game";
 import { DefaultBoardConfig, DefaultBoardState } from "./lib/board_types";
 
+/**
+ * Describes a game variant, including its configuration, behavior, and UI representation.
+ *
+ * Each variant defines how the game logic is handled (`gameClass`), how players interact with it,
+ * and how the game is presented in the UI. This interface is used to register and manage different
+ * types of games within the platform.
+ *
+ * @template ConfigT - The shape of the configuration object used to initialize the game.
+ * @template StateT - The shape of the state object used to represent ongoing gameplay.
+ */
 export interface Variant<
   ConfigT extends object = object,
   StateT extends object = object,
 > {
   /** The class that implements AbstractGame and manages the state of an ongoing game */
   gameClass: new (config: ConfigT) => AbstractGame;
+  /** Short description of the Variant (shown to players in-game) */
   description: string;
+  /**
+   * Long description of the variant.  Supports markdown syntax.
+   * shown at /variants/<variant>/rules
+   */
   rulesDescription?: string;
+  /**
+   * Set this field to true to mark the variant as deprecated. Existing games (ongoing or
+   * completed) will remain accessible, but new games cannot be created.
+   */
   deprecated?: boolean;
+  /** Used to determine the flow of the game.  In most cases, this is "sequential". */
```

Comment:
Maybe we could say that parallel means that there rounds where multiple players have a decision to make.

Reply:
Thanks for the feedback!  Added descriptions of each option.